internal/app/concurrent_proxy_stage_test.go

```diff
@@ -199,8 +212,12 @@ func (s *ConcurrentProxyStage) all_the_user_responses_should_have_the_right_stat
 	expectedLen := s.concurrentUserRequestsPerSecond * int(s.concurrentUserRequestsDuration/time.Second)
 	s.assert.Len(s.userResponses, expectedLen, "number of user responses is not as expected")
 
-	for _, res := range s.userResponses {
-		s.assert.Equal(res.StatusCode, s.modifiedNameStatusCode, "expected user status code")
+	for i, res := range s.userResponses {
+		if s.modifiedNameAttempt == nil || *s.modifiedNameAttempt == i+1 {
```

Comment:
Modifying attempt and number of concurrent requests have different results here - runs are concurrent so it may be out of order, perhaps the test would be more valuable if window of opportunity to fail was greater like using another fixture in test for example 10 requests per second and modify on attempt 4, or 7.

you'd need to figure a way to identify request with response in another way though